lib/retry.go

```diff
@@ -9,16 +9,53 @@ const (
 	defaultMaxWait     = 2 * time.Minute
 )
 
+// Interface used for offloading jitter calculations from the RetryWaiter
+type Jitter interface {
+	AddJitter(baseTime time.Duration) time.Duration
+}
+
+// Calculates a random jitter between 0 and up to a specific percentage of the baseTime
+type JitterRandomStagger struct {
+	// int64 because we are going to be doing math against an int64 to represent nanoseconds
+	percent int64
+}
+
+// Creates a new JitterRandomStagger
+func NewJitterRandomStagger(percent int) *JitterRandomStagger {
+	if percent < 0 {
+		percent = 0
+	} else if percent > 100 {
```

Comment:
nit: more than 100 seems reasonable too - might want to add multiples of the base time like 200% but probably unlikely for our usage.